src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,117 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String NEGATIVE_INDEX_MESSAGE = "Index must be positive number";
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "Index must be positive number";
+    private final ArrayList<Node<T>> data = new ArrayList<>();
+    private int size;
+
+    private static class Node<V> {
+        private V value;
+        private Node<V> next;
+        private Node<V> prev;
+
+        Node(Node<V> prev, V value, Node<V> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        data.add(new Node<>((size == 0) ? null : data.get(size - 1), value, null));
+        if (size > 0) {
+            data.get(size - 1).next = data.get(size);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexValidator(index, false);
+        data.add(index, new Node<>(
+                (index > 0) ? data.get(index - 1) : null,
+                value,
+                (size > 0 && index < size - 1) ? data.get(index) : null));
+        if (index > 0) {
+            data.get(index - 1).next = data.get(index);
+        }
+        if (index < size) {
+            data.get(index + 1).prev = data.get(index);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int newSize = size + list.size() * 2;
+        for (int i = size; i < newSize; i = i + 2) {
+            add(list.get(i - size));
+        }
     }
```

Comment:
The addAll method should not double the size of the list when adding elements. It can simply iterate over the list parameter and call the add method for each element.

Reply:
addAll() doesn't doubles list's size. newSize - variable for cycle's process limiting, that's all